lib/spack/spack/concretize.py

```diff
@@ -161,7 +161,8 @@ def concretize_compiler(self, spec):
             return False
 
         try:
-            nearest = next(p for p in spec.traverse(direction='parents')
+            nearest = next(p for p in spec.traverse(direction='parents',
+                                                    deptype=(spack.DEP_BUILD, spack.DEP_INSTALL))
```

Comment:
what's the difference between `deptype` and `deptype_query`?  shouldn't external code use only one of these?